### ISSUE TITLE
Discussion Item: Signaling of DRM information via ServiceAccessInformation

### ISSUE THREAD

**Issue by dsilhavy** [5G-MAG/Standards#49]
# Problem description
TS 26.512 defines a way to signal the MPD URL as part of the `ServiceAccessInformation` resource. For that reason, the `StreamingAccess.mediaPlayerEntry` field can be used. To my best knowledge there is no similar mechanism to signal the corresponding DRM license server and DRM custom headers via the `ServiceAccessInformation` resource. Consequently, this information needs to be embedded in the DASH manifest or the DASH segments or provided via a different interface.

While a Playready DRM allows signaling the license server URL using the `pssh` box of the segments, this is not possible for Widevine.

For manifest files the DASH-IF Content Protection guidelines define the `dashif:Laurl` element to specify the license server:

````
<dashif:Laurl>https://example.com/AcquireLicense</dashif:Laurl>
````

However, this might not be supported by all media players, for instance I did not find any usage of the `dashif:Laurl` element in Exoplayer.

In addition, the DASH-IF Content Protection guidelines define the `dashif:Authzurl` element to retrieve  authorization tokens that may be required for requesting a license from a license server:

```` 
<dashif:Authzurl>https://example.com/tenants/5341/authorize</dashif:Authzurl>
```` 

Again client support might be limited.

# Suggested solution description
This is more a suggestion to discuss if it could be useful to signal DRM information as part of the `ServiceAccessInformation` resource. That would allow applications to pass the required parameters to the media player instead of expecting the media player to derive the required information from the manifest or the segments. In addition, it would allow for changes of the DRM information without touching the manifest or the segments.

**Comment by rjb1000** [comment]
My opinion on this is that DRM is beyond the scope of 3GPP standardisation. The DRM client does not appear as an actor in the 5GMS architecture described in TS 26.501. DRM licence acquisition doesn't appear in the generic procedural call flow in figure 5.1-1. It appears for the first time in figure 5.2-2 and later in 5.7-2, both of which are depicting the 5G Media Streaming architecture in the context of DASH streaming. The DRM licence acquisition is shown as coming from the Media Player.

My conclusion is that either:
- The DRM licence server is referenced from the MPD, as you describe above, or
- It is the application's responsibility to configure the DRM client privately.

I don't think it would be appropriate to include DRM information in the 5GMS Service Access Information.

**Comment by rjb1000** [comment]
Action: @haudiobe to discuss with @dsilhavy.

**Comment by dsilhavy** [comment]
Thanks for the feedback @rjb1000 . I would be interested to hear @haudiobe opinion on that as well.

What I find irritating is that we add the URL to the MPD to the `ServiceAccessInformation`. However, for playback of DRM protected content this is information is not sufficient in most cases as described above. From a client's perspective, I am unable to play this content without the URL to the DRM server and potential DRM headers. Consequently, I would argue that the `ServiceAccessInformation` is incomplete as I can't access the service without information provided by the application provider (probably via M8?)

From `26.501 4.2.3`: _The Service Access Information is the set of parameters and addresses which are needed by the 5GMSd Client to activate and control the reception of a downlink streaming session, and to report service/content consumption and/or QoE metrics_

I guess it comes down to the definition of _activate_ and _control_.

**Comment by rjb1000** [comment]
Reassigned to Rel-19 label reflecting the likelihood that this topic will be studied in the Advanced Media Delivery feasibility study which was agreed at SA4#127 (Sophia Antipolis). The Work Item Description is to be considered for approval at SA#103 (Maastrict).

**Comment by jordijoangimenez** [comment]
@haudiobe to check status

**Comment by haudiobe** [comment]
> Thanks for the feedback [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/9865771?u=2bcb1864210c5b6a88e19e01ce48c4404eb14174&amp;v=4&amp;size=80">@rjb1000](https://github.com/rjb1000) . I would be interested to hear [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/2828689?s=64&amp;u=244b9e5ab489412a4f65b72183e85c5bf23a1285&amp;v=4">@haudiobe](https://github.com/haudiobe) opinion on that as well.
> 
> What I find irritating is that we add the URL to the MPD to the `ServiceAccessInformation`. However, for playback of DRM protected content this is information is not sufficient in most cases as described above. From a client's perspective, I am unable to play this content without the URL to the DRM server and potential DRM headers. Consequently, I would argue that the `ServiceAccessInformation` is incomplete as I can't access the service without information provided by the application provider (probably via M8?)
> 
> From `26.501 4.2.3`: _The Service Access Information is the set of parameters and addresses which are needed by the 5GMSd Client to activate and control the reception of a downlink streaming session, and to report service/content consumption and/or QoE metrics_
> 
> I guess it comes down to the definition of _activate_ and _control_.

Why would the inclusion of the DRM information in the Media Player Entry not be sufficient? The MPD can carry Content Protection information and I expect HLS can do the same.

**Comment by dsilhavy** [comment]
> > Thanks for the feedback [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/9865771?u=2bcb1864210c5b6a88e19e01ce48c4404eb14174&amp;v=4&amp;size=80">@rjb1000](https://github.com/rjb1000) . I would be interested to hear [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/2828689?s=64&amp;u=244b9e5ab489412a4f65b72183e85c5bf23a1285&amp;v=4">@haudiobe](https://github.com/haudiobe) opinion on that as well.
> > What I find irritating is that we add the URL to the MPD to the `ServiceAccessInformation`. However, for playback of DRM protected content this is information is not sufficient in most cases as described above. From a client's perspective, I am unable to play this content without the URL to the DRM server and potential DRM headers. Consequently, I would argue that the `ServiceAccessInformation` is incomplete as I can't access the service without information provided by the application provider (probably via M8?)
> > From `26.501 4.2.3`: _The Service Access Information is the set of parameters and addresses which are needed by the 5GMSd Client to activate and control the reception of a downlink streaming session, and to report service/content consumption and/or QoE metrics_
> > I guess it comes down to the definition of _activate_ and _control_.
> 
> Why would the inclusion of the DRM information in the Media Player Entry not be sufficient? The MPD can carry Content Protection information and I expect HLS can do the same.

I agree that it is possible to carry DRM information such as the license server URL in the MPD. However, something that is missing today is the ability to signal a certificate server. If we expect all relevant DRM parameters to be signaled via the manifest or provided via the application, we can close this issue. 

I am adding some context from a mail that @tobbee sent to the CPS group:

> Livesim2 supports encrypting content on the fly and providing relevant DASH-IF Laurl elements in the MPD.
For example:
https://livesim2.dashif.org/livesim2/drm_EZDRM-1-key-cbcs/testpic_2s/Manifest.mpd
contains the

````
<dashif:Laurl xmlns:dashif=https://dashif.org/CPS licenseType="EME-1.0">https://playready.ezdrm.com/cency/preauth.aspx?pX=274FF4</dashif:Laurl>
````
> However, for some DRM systems, there is also a need to specify a certificate server. This can currently be specified in dash.js and shaka-player as a configuration parameter for that asset in a list of test assets, but it would be much nicer for testing with dash.js if we could signal that in the URL without needing a side channel.
> 
> Something like:
> <dashif:certificate xmlns:dashif="https://dashif.org/CPS">CERTIFICATE_URL</dashif:certificate>
> should be fine, I think.
> 
> Do you see any issues with adding this to our DASH-IF namespace?
>
